docs/reference/files/file-charmcraft-yaml.rst

```diff
@@ -0,0 +1,901 @@
+
+``charmcraft.yaml`` is a file in your charm project that contains keys that allow you to declare information about the project in a form that can be used by Charmcraft.
+
+.. note::
+    If you're starting from an empty file, the only required key is the ``type`` key. However, depending on what value you set it to (``charm`` or ``bundle``), other keys become required as well.
+
+
+.. dropdown:: Expand to view a full charm with sample content all at once
+
+    .. literalinclude:: charmcraft-sample-charm.yaml
+
+.. _file-charmcraft-yaml-actions::
+``actions``
+===========
+
+    Definition owned by `Juju`_. Used by `Charmcraft`_, `Charmhub`_, `Juju`_.
+
+**Status:** Optional.
+
+**Purpose:** Defines actions the charm can take.
+
+**Name:** String = user-defined action name.
+
+**Value:** Mapping.
+
+The value of this key is the contents of :ref:`file-actions-yaml`.
+
+.. dropdown:: Example
+
+    .. literalinclude:: charmcraft-sample-charm.yaml
+        :start-at: actions:
+        :end-before: analysis:
+
+
+.. _file-charmcraft-yaml-analysis::
+``analysis``
+============
+
+    Definition owned by `Charmcraft`_. Used by `Charmcraft`_.
+
+**Status:** Optional.
+
+**Purpose:** Defines how the analysis done on the package will behave. This analysis is run implicitly as part of the ``pack`` command but can be called explicitly with the ``charmcraft analyse`` command.
+
+**Structure:**
+
+.. code::
+
+    analysis:
+      ignore:
+        attributes: [<check-name>,...]
+        linters: [<check-name>,...]
+
+.. dropdown:: Example
+
+    .. literalinclude:: charmcraft-sample-charm.yaml
+        :start-at: analysis:
+        :end-before: assumes:
+
+.. _file-charmcraft-yaml-assumes::
+``assumes``
+===========
+
+**Status:** Optional. Recommended for Kubernetes charms.
+
+**Purpose:** Allows charm authors to explicitly state in the metadata of a charm various features that a Juju model must be able to provide to ensure that the charm can be successfully deployed on it. When a charm comes preloaded with such requirements, this enables Juju to perform a pre-deployment check and to display user-friendly error messages if a feature requirement cannot be met by the model that the user is trying to deploy the charm to. If the assumes section of the charm metadata is omitted, Juju will make a best-effort attempt to deploy the charm, and users must rely on the output of `juju status` to figure out whether the deployment was successful. The `assumes` key is available since 2.9.23.
+
+**Structure:** The key consists of a list of features that can be given either directly or, depending on the complexity of the condition you want to enforce, nested under one or both of the boolean expressions ``any-of`` or ``all-of``, as shown below. In order for a charm to be deployed, all entries in the ``assumes` block must be satisfied.
```

Comment:
Pandoc's fault?

```suggestion
**Structure:** The key consists of a list of features that can be given either directly or, depending on the complexity of the condition you want to enforce, nested under one or both of the boolean expressions ``any-of`` or ``all-of``, as shown below. In order for a charm to be deployed, all entries in the ``assumes`` block must be satisfied.
```

Reply:
A combination of pandoc and alexdoc (pandoc left most of these as single-backticks and I missed that. Thanks for catching it!